_source/_posts/2020-06-22-deploy-dotnet-container-aws-fargate.md

```diff
@@ -3,112 +3,99 @@ disqus_thread_id: 8089923701
 
-## Understanding Containers vs Virtual Machines
+{% include toc.md %}
 
-Around 2013, a product was released called [Docker](https://www.docker.com/), and with it began a shift of how we think about hosting applications and managing infrastructure. Infrastructure teams and operations have been leveraging Virtual Machines with great success for decades, so you may be tempted to think "isn't this just a virtual machine?" In spirit, yes, in application no.
+## Understanding Containers vs. Virtual Machines
 
-To talk about containers we have to talk about Virtual Machines. A virtual machine is a virtual representation of a computer all the way from its boot process to loading a full operating system. This provides a lot of flexibility because you get to have fine-grained control over an entire virtual computer, but it has all the pitfalls of a full computer as well. The Operating System has to be maintained, patched, and updated; disks have to be managed, and all manner of operational overhead goes into their care and feeding. Notably, they are also very large because of the fact that an entire OS has to go on them. So when you start a Virtual Machine you have to go through the entire boot sequence of a traditional computer.
+Around 2013, [Docker](https://www.docker.com/) was released, and with it began a shift in how we think about hosting applications and managing infrastructure. Infrastructure teams and operations have been leveraging Virtual Machines with great success for decades, so you may be tempted to think, "isn't this just a virtual machine?" In spirit, yes, in application no.
 
-These are the 3 components that containers hope to solve. Ease of management, size, and speed to start. There are others but for simplicity, we'll focus on those 3 as I feel (IMHO) they are the most relevant to business processes. Containers differ from a traditional virtual machine in one major way. They do not have an actual Operating System. This may come as a surprise to you, even if you might have dabbled with containers before. Containers instead have an abstraction of an operating system that provides hooks into the host operating system using standards and conventions supplied by the container system.
+To talk about containers, we have to talk about Virtual Machines. A virtual machine is a virtual representation of a computer all the way from its boot process to loading an entire operating system. This provides a lot of flexibility because you get to have fine-grained control over an entire virtual computer, but it has all the pitfalls of a whole computer as well. The Operating System has to be maintained, patched, and updated; disks have to be managed, and all manner of operational overhead goes into their care and feeding. Notably, they are also huge because an entire OS has to go on them. So when you start a Virtual Machine, you have to go through the entire boot sequence of a traditional computer.
```

Comment:
```suggestion
To talk about containers, we have to talk about Virtual Machines. A virtual machine (VM) is a virtual representation of a computer all the way from its boot process to loading an entire operating system. This provides a lot of flexibility because you gain fine-grained control over an entire virtual computer, but it brings all the pitfalls of a whole computer as well. The operating system has to be maintained, patched, and updated; disks have to be managed, and all manner of operational overhead goes into their care and feeding. Notably, VMs are huge because an entire OS has to go on them. So when you start a virtual machine, you have to go through the entire boot sequence of a traditional computer.
```